llvm/lib/Target/AArch64/AArch64FrameLowering.cpp

```diff
@@ -2590,9 +2590,11 @@ StackOffset AArch64FrameLowering::resolveFrameOffsetReference(
         // via the frame pointer, so we have to use the FP in the parent
         // function.
         (void) Subtarget;
+        // swifttailcc uses FP
         assert(
-            Subtarget.isCallingConvWin64(MF.getFunction().getCallingConv()) &&
-            "Funclets should only be present on Win64");
+            (Subtarget.isCallingConvWin64(MF.getFunction().getCallingConv()) ||
+             MF.getFunction().getCallingConv() == CallingConv::SwiftTail) &&
+            "Funclets should only be present on Win64 or swifttailcc");
```

Comment:
What happens if I have a `swiftcc` function? Is there a reason that loosening the assertion like this is appropriate?

Reply:
swiftcc is already covered by isCallingConvWin64. I think this is checking that the particular cc uses a frame pointer, which swifttailcc does, and being relatively a recent addition, it wasn't covered by this code.

Reply:
Ah, okay!  That I think is relevant details that should be in the commit message.

Reply:
Done